crates/youki/src/logger.rs

```diff
@@ -93,3 +103,55 @@ where
 
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+    use serial_test::serial;
+
+    use super::*;
+    use std::env;
+    struct LogLevelGuard {
+        original_level: Option<String>,
+    }
+
+    impl LogLevelGuard {
+        fn new(level: &str) -> Result<Self> {
+            let original_level = env::var(LOG_LEVEL_ENV_NAME).ok();
+            env::set_var(LOG_LEVEL_ENV_NAME, level);
+            Ok(Self { original_level })
+        }
+    }
+    impl Drop for LogLevelGuard {
+        fn drop(self: &mut LogLevelGuard) {
+            if let Some(level) = self.original_level.as_ref() {
+                env::set_var(LOG_LEVEL_ENV_NAME, level);
+            } else {
+                env::remove_var(LOG_LEVEL_ENV_NAME);
+            }
+        }
+    }
+
+    #[test]
+    fn test_detect_log_level_is_debug() {
+        let _guard = LogLevelGuard::new("error").unwrap();
+        assert_eq!(detect_log_level(true).unwrap(), LevelFilter::Debug)
+    }
+
+    #[test]
+    #[serial]
+    fn test_detect_log_level_default() {
+        env::remove_var(LOG_LEVEL_ENV_NAME);
```

Comment:
Do you also want to take advantage of the LogGuard here?
1. remember the current value
2. remove the current value
3. afterwards, restore the previous value

Make `new` take in Option<log_level>.

Reply:
It is important that the environment variable is not set in this test, but it looks like we can set the original environment variable by using the guard after this test. Let's do it.